addons/html_editor/static/src/editor/core/dom_plugin.js

```diff
@@ -25,6 +25,15 @@ export class DomPlugin extends Plugin {
             case "INSERT_SEPARATOR":
                 this.insertSeparator();
                 break;
+            case "CLEAN": {
+                const root = payload;
+                for (const node of [root, ...descendants(root)]) {
```

Comment:
*_Note/reflection for later_*
To avoid traversing the DOM twice here maybe we should have a version of `descendants` with a callback to apply the cleaning as we go, although we'd have to be careful not to introduce side effects. It might be worth it in a very large document, it's an open question.
Something like:
```js
iterateOverDescendants(root, { includeRoot: true }, node => {
    if (!node.classList.length) {
        node.removeAttribute('class');
    }
});
```

Reply:
The cleaning (we can change the term to something else) is what should be done before saving.
What do you mean by avoid traversing the DOM twice ?